CRM/Extension/Browser.php

```diff
@@ -157,21 +157,31 @@ public function getExtensions() {
 
   /**
    * Get a description of a particular extension.
+   * This function is an optimised form of getExtensions() - it returns the
+   * extension's details as soon as it's found without caching anything.
    *
    * @param string $key
    *   Fully-qualified extension name.
    *
    * @return CRM_Extension_Info|NULL
    */
   public function getExtension($key) {
-    // TODO optimize performance -- we don't need to fetch/cache the entire repo
-    $exts = $this->getExtensions();
-    if (array_key_exists($key, $exts)) {
-      return $exts[$key];
+    if (!$this->isEnabled() || count($this->checkRequirements())) {
+      return NULL;
     }
-    else {
+
+    $remote = $this->_discoverRemote();
```

Comment:
Mind you, (taking a second look at it - thanks for feedback!) as the next phase of this, one could apply the same theory (returning the extension when found and not caching) for the _discoverRemote function. That would optimise it still more...

I may consider giving that a go before seeking approval for this PR, as that might be faster still.